Gems/Atom/RPI/Code/Source/RPI.Public/Pass/PassSystem.cpp

```diff
@@ -357,19 +290,30 @@ namespace AZ
             return m_rootPass;
         }
 
-        PassSystemState PassSystem::GetState() const
+        void PassSystem::AddRenderPipeline(RenderPipeline* renderPipeline)
         {
-            return m_state;
+            m_renderPipelines.push_back(renderPipeline);
+            m_rootPass->AddChild(renderPipeline->m_passes.m_rootPass);
         }
 
-        bool PassSystem::IsHotReloading() const
+        void PassSystem::RemoveRenderPipeline(RenderPipeline* renderPipeline)
         {
-            return m_isHotReloading;
+            renderPipeline->m_passes.ProcessQueuedChanges();
+            renderPipeline->m_passes.m_rootPass->SetEnabled(false);
+            renderPipeline->m_passes.m_rootPass->QueueForRemoval();
+            renderPipeline->m_passes.ProcessQueuedChanges();
+
+            erase(m_renderPipelines, renderPipeline);
         }
 
-        void PassSystem::SetHotReloading(bool hotReloading)
+        void PassSystem::AddNonePipelinePass(const Ptr<Pass>& pass)
+        {
+            m_passesWithoutPipeline.m_rootPass->AddChild(pass);
+        }
```

Comment:
nit: not a fan of this function name, I need to understand the implementation to understand the name.

Suggestion: AddPassWithoutPipeline

Reply:
Sure, sounds good, will change